src/main/java/at/hannibal2/skyhanni/features/misc/discordrpc/DiscordStatus.kt

```diff
@@ -137,6 +141,61 @@ enum class DiscordStatus(private val displayMessageSupplier: Supplier<String>?)
 
     CUSTOM({
         SkyHanniMod.feature.misc.discordRPC.customText.get() // custom field in the config
+    }),
+
+    AUTO({
+        val slayerResult = SLAYER.displayMessageSupplier!!.get()
+        val milestoneResult = try {
+            CROPMILESTONES.displayMessageSupplier!!.get()
+        }
+        catch (e: Exception){
+            "Unable to get milestone"
+        }
+        if (slayerResult != "Planning to do a slayer quest") slayerResult
+        else if (milestoneResult != "Unable to get milestone" && milestoneResult != "Unknown Item") milestoneResult
+        else {
+            val statusNoAuto = DiscordStatus.values().toMutableList()
+            statusNoAuto.removeAt(10)
+            statusNoAuto[SkyHanniMod.feature.misc.discordRPC.auto.get()].getDisplayString()
+        }
+    }),
+
+    CROPMILESTONES({
+        val item = net.minecraft.client.Minecraft.getMinecraft().thePlayer.heldItem
+        val crop = item.getCropType()
+        val cropCounter = crop?.getCounter()
+        val tier = cropCounter?.let { getTierForCrops(it) }
+
+        val progress = if (cropCounter != null && tier != null) {
+            val currentTierOffset = getCropsForTier(tier).toFloat()
+            val calc = 100 * ((cropCounter.toFloat() - currentTierOffset) / (getCropsForTier(tier + 1).toFloat() - currentTierOffset))
```

Comment:
Somewhere in CropMilestoneDisplay we already calculate the percentage of the current crop. Maybe move that into a utils method and use this method here instead